lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -5150,11 +5148,16 @@ impl<'a> FuncGen<'a> {
             Operator::Call { function_index } => {
                 self.emit_call(FunctionIndex::from_u32(function_index))?
             }
-            Operator::CallIndirect { index, table_index } => {
+            Operator::CallIndirect {
+                type_index,
+                table_index,
+                table_byte,
+            } => {
                 // TODO: removed restriction on always being table idx 0;
                 // does any code depend on this?
+                let _table_byte = table_byte; // TODO: neither wasmparser docs nor the wasm core spec define this
```

Comment:
I believe this is basically just an artifact of how the webassembly specification has evolved.

In particular the wasm-core-1 spec binary encoding for `call_indirect` is as thus:

```
0x11  x:typeidx  0x00  =>  call_indirect x​
```

Here both `0x11` and `0x00` are singular _bytes_. And the current specification (post `reference_types`) has:

```
0x11  x:typeidx  x:tableidx  =>  call_indirect x​
```

where `tableidx` is a LEB-encoded u32. LEB-encoding of `0u32` is the same `0x00`, but LEB `0x80 0x00` is _also_ a `0u32`. So unless `wasmparser` carries around the first byte, they can’t check for the version of the specification/its extensions (which in turn happens in the validator, rather than the parser.)

With that in mind we should probably just `table_byte: _` above in the match.

Reply:
Oh ok I see, thank you for the explanation!